### PR TITLE
style: add missing spinners to `flox {update,upgrade,push}`

### PR DESCRIPTION
**Pull request by ysndr**
Adds spinning messages to more potentially long-running tasks.

### CODE REVIEW

File: cli/flox/src/commands/environment.rs [hunk 1]
```diff
@@ -1426,8 +1426,14 @@ impl Push {
             EnvironmentPointer::Managed(managed_pointer) => {
                 let message = Self::push_existing_message(&managed_pointer, self.force);
 
-                // todo add spinner
-                Self::push_managed_env(&flox, managed_pointer, dir, self.force)?;
+                Dialog {
+                    message: "Pudhing updates to FloxHub...",
```
Comment (mkenigs) [inline cli/flox/src/commands/environment.rs:1430]:
```suggestion
                    message: "Pushing updates to FloxHub...",
```

File: cli/flox/src/commands/environment.rs [hunk 2]
```diff
@@ -2207,7 +2219,14 @@ impl Upgrade {
 
         let mut environment = concrete_environment.into_dyn_environment();
 
-        let upgraded = environment.upgrade(&flox, &self.groups_or_iids)?.packages;
+        let result = Dialog {
+            message: "Upgrading packages and verifying environment...",
```
Comment (mkenigs) [inline cli/flox/src/commands/environment.rs:2223]: nonblocking
```suggestion
            message: "Upgrading packages...",
```

**Review by mkenigs — CHANGES_REQUESTED** [review]
nonblocking: we should add one for the first half of `update` too (there's one for the scrape, but not bumping and locking)

File: cli/flox/src/commands/environment.rs (hunk 1 above)
Comment (ysndr) [reply cli/flox/src/commands/environment.rs:1430]: 🍮

File: cli/flox/src/commands/environment.rs [hunk 3]
```diff
@@ -2055,7 +2067,12 @@ impl Update {
                     new_lockfile,
                     old_lockfile,
                     ..
-                } = self.update_manifest(flox, concrete_environment)?;
+                } = Dialog {
+                    message: "Updating environment...",
+                    help_message: None,
+                    typed: Spinner::new(|| self.update_manifest(flox, concrete_environment)),
+                }
```
Comment (mkenigs) [inline cli/flox/src/commands/environment.rs:2074]: nonblocking: I think we should have one on `LockedManifest::update_global_manifest(&flox, self.inputs)` as well

**Review by mkenigs — APPROVED** [review]

**Review by mkenigs — APPROVED** [review]

**Pull request merged**